File: app/Sources/MunkelApp/GitHubConfig.swift
import Foundation

enum GitHubConfig {
    /// OAuth app "Munkel" (org limehq, device flow enabled). Client IDs
    /// are public by design — the device flow needs no secret.
    static let defaultClientID = "Ov23liAGgt648fdFmImz"

    /// Overridable without recompiling:
    /// `defaults write dev.uq.munkel githubClientID <CLIENT_ID>`
    static var clientID: String {
        UserDefaults.standard.string(forKey: "githubClientID") ?? defaultClientID
    }

    static var isConfigured: Bool { !clientID.isEmpty }
}

File: apps/windows/src/main/github-config.ts
const DEFAULT_CLIENT_ID = 'Ov23liAGgt648fdFmImz';

// public OAuth client ID = same as macOS GitHubConfig.swift; device flow needs no secret; override via MUNKEL_GITHUB_CLIENT_ID
export function getGitHubClientID(): string {
	const override = process.env.MUNKEL_GITHUB_CLIENT_ID?.trim();
	return override && override.length > 0 ? override : DEFAULT_CLIENT_ID;
}
